src/components/dialogs/showUsers/ShowUsersDialog.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {useState, useEffect} from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import { DialogTitle } from '../addTodo/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import TableFooter from '@material-ui/core/TableFooter';
+import Paper from '@material-ui/core/Paper';
+import { routes, pagination } from '../../../config/constants';
+import { Transition } from '../addTodo/styles';
+import { UsersData } from '../../../utils/interfaces/users';
+import { TodoCheckbox } from '../../checkbox/TodoCheckbox';
+import { TodoPagination } from '../../pagination/TodoPagination';
+import axios from 'axios';
+
+interface Props {
+  open: boolean;
+  closeDialog: () => void;
+}
+
+export const ShowUsersDialog = ({ open, closeDialog }: Props) => {
+  const [ users, setUsers ] = useState<UsersData[]>([]);
+  const [ todosNumber, setTodosNumber ] = useState<number[]>([]);
+  const [ allUsersCount, setAllUsersCount ] = useState<number>(0);
+
+  const [ isOpen, setOpen ] = useState<boolean>(open);
+  const [ chosenUsers, setChosenUsers ] = useState<number[]>([]);
+
+  const [ currentPer, setPer ] = useState<number>(pagination.rowsOnPage[0]);
+  const [ currentPage, setPage ] = useState<number>(1);
+
+  const getUsers = (newPer: number, newPage: number) => {
+    axios.get(`${routes.server}/${routes.users}`, {
+      params: {
+        per: newPer,
+        page: newPage
+      }
+    }).then(res => {
+      setUsers(res.data.users);
+      setTodosNumber(res.data.todos_number);
+      setAllUsersCount(res.data.total_record_count);
+    });
+  }
+
+  const deleteUsers = () => {
+    axios.delete(`${routes.server}/${routes.users}`, { data: {
+      ids: chosenUsers
+    }}).then(res => {
+      setUsers(res.data.users);
+      setTodosNumber(res.data.todos_number);
+      setAllUsersCount(res.data.total_record_count);
```

Comment:
Sta prosto kara ini

`res.data.users.length`

Reply:
`res.data.users`-y en usernenn, verunq hamapatasxanumn tvyal page-in ev per-in. Isk `res.data.total_record_count`-y sax userneri qanakna. Ozyum asim `res.data.users`-i lengthy karoxa 5 ini (entadrinq min egum amenashaty karoxa 5 todo ini), isk `res.data.total_record_count`-y 15 (yndameny 15 user ka)

Reply:
Lav et qez loxchin county petqa apa sta lav varianta:+1: